provider/swarm.go

```diff
@@ -0,0 +1,332 @@
+		"getPort":           provider.getPort,
+		"getWeight":         provider.getWeight,
+		"getDomain":         provider.getDomain,
+		"getProtocol":       provider.getProtocol,
+		"getPassHostHeader": provider.getPassHostHeader,
+		"getPriority":       provider.getPriority,
+		"getEntryPoints":    provider.getEntryPoints,
+		"getFrontendRule":   provider.getFrontendRule,
+		"replace":           replace,
+	}
+
+	// filter services
+	filteredServices := fun.Filter(func(service swarm.Service) bool {
+		return provider.serviceFilter(service, provider.ExposedByDefault)
+	}, servicesInspected).([]swarm.Service)
+
+	frontends := map[string][]swarm.Service{}
+	for _, service := range filteredServices {
+		frontendName := provider.getFrontendName(service)
+		frontends[frontendName] = append(frontends[frontendName], service)
+	}
+
+	templateObjects := struct {
+		Services  []swarm.Service
+		Frontends map[string][]swarm.Service
+		Domain    string
+	}{
+		filteredServices,
+		frontends,
+		provider.Domain,
+	}
+
+	configuration, err := provider.getConfiguration("templates/swarm.tmpl", DockerFuncMap, templateObjects)
+	if err != nil {
+		log.Error(err)
+	}
+	return configuration
+}
+
+func (provider *Swarm) serviceFilter(service swarm.Service, exposedByDefaultFlag bool) bool {
+	_, err := strconv.Atoi(service.Spec.Annotations.Labels["traefik.port"])
+
+	if err != nil {
+		log.Debugf("Filtering service without traefik.port label %s", service.Spec.Annotations.Name)
+		return false
+	}
+
+	if !isServiceEnabled(service, exposedByDefaultFlag) {
+		log.Debugf("Filtering disabled service %s", service.Spec.Annotations.Name)
+		return false
+	}
+
+	serviceTags := strings.Split(service.Spec.Annotations.Labels["traefik.tags"], ",")
+	if ok, failingConstraint := provider.MatchConstraints(serviceTags); !ok {
+		if failingConstraint != nil {
+			log.Debugf("Service %v pruned by '%v' constraint", service.Spec.Annotations.Name, failingConstraint.String())
+		}
+		return false
+	}
+
+	return true
+}
+
+func (provider *Swarm) getFrontendName(service swarm.Service) string {
+	// Replace '.' with '-' in quoted keys because of this issue https://github.com/BurntSushi/toml/issues/78
+	return normalize(provider.getFrontendRule(service))
+}
+
+// GetFrontendRule returns the frontend rule for the specified container, using
+// it's label. It returns a default one (Host) if the label is not present.
+func (provider *Swarm) getFrontendRule(service swarm.Service) string {
+	if label, err := getServiceLabel(service, "traefik.frontend.rule"); err == nil {
+		return label
+	}
+	return "Host:" + provider.getSubDomain(service.Spec.Annotations.Name) + "." + provider.Domain
+}
+
+func (provider *Swarm) getBackend(service swarm.Service) string {
+	if label, err := getServiceLabel(service, "traefik.backend"); err == nil {
+		return label
+	}
+	return normalize(service.Spec.Annotations.Name)
+}
+
+func (provider *Swarm) getIPAddress(service swarm.Service) string {
```

Comment:
ResolutionModeDNSRR: Traefik is a load balancer and then have to manage routing by himself. A DNS round robin rules is not expected in the Swarm provider. ;-)

ResolutionModeVIP: This loop will always return the ip of the first container registered in the service. It is also not expected :trollface:

I think you should get the container list in the ```Provide``` function and add every containers (=endpoints) in a single slice, whatever the service they are executing. Then loop over this slice as you did with services.